### ISSUE TITLE
Local extensions not showing proper error reporting

### ISSUE THREAD

**Issue by RuudBurger** [bolt/bolt#2652]
I get a white screen when there is a php error in my extension. This makes writing code pretty difficult.
Looking at https://github.com/bolt/bolt/blob/master/src/Extensions.php#L171-L178 it just ignores any error in the autoload initialize() function.

**Comment by GawainLynch** [comment]
Testing here and I am getting proper exceptions/Whoops.  Are you testing on a site that you're not logged into?  

By default we don't show these errors to non logged in users. But you can set `debug_show_loggedoff: true` in your `config.yml` to make them show.

Regardless I have added a log call to the two try/catch blocks.

**Comment by RuudBurger** [comment]
debug_show_loggedoff is on true (but I'm logged in as root anyway).
I do get all other errors I make in twig (calling non existing function to test).
I am using composer (which seems to be important in all the questions I'm having ;))

My extension
```php
<?php

	namespace Bolt\Extension\Company\Async;

	use Bolt\Application;
	use Bolt\BaseExtension;

	class Extension extends BaseExtension
	{
		public function initialize()
		{
			$this->test(); // I'm expecting an error for this
		}

		public function getName()
		{
			return "async";
		}
	}

```

And I see this stack trace in php logs:
```
[01-Feb-2015 09:07:39 UTC] PHP Stack trace:
[01-Feb-2015 09:07:39 UTC] PHP   1. {main}() /Users/Ruud/Localhost/Site/public/index.php:0
[01-Feb-2015 09:07:39 UTC] PHP   2. Bolt\Application->initialize() /Users/Ruud/Localhost/Site/public/index.php:18
[01-Feb-2015 09:07:39 UTC] PHP   3. Bolt\Application->initExtensions() /Users/Ruud/Localhost/Site/vendor/bolt/bolt/src/Application.php:110
[01-Feb-2015 09:07:39 UTC] PHP   4. Bolt\Extensions->initialize() /Users/Ruud/Localhost/Site/vendor/bolt/bolt/src/Application.php:314
[01-Feb-2015 09:07:39 UTC] PHP   5. Bolt\Extensions->initializeExtension() /Users/Ruud/Localhost/Site/vendor/bolt/bolt/src/Extensions.php:223
[01-Feb-2015 09:07:39 UTC] PHP   6. Bolt\Extension\Company\Async\Extension->initialize() /Users/Ruud/Localhost/Site/vendor/bolt/bolt/src/Extensions.php:242
```
config_local.yml
```yaml
debug: true
debug_error_level: -1
debug_show_loggedoff: true
debug_enable_whoops: true
```

**Comment by RuudBurger** [comment]
Ok something else. If I do a `var_dump('test')` in the initialize. I also get a blank screen (with `test` in it). So it seems to not continue loading. Php stack change from the logs:
```
01-Feb-2015 09:18:44 UTC] PHP Fatal error:  Uncaught exception 'RuntimeException' with message 'Failed to start the session because headers have already been sent by "/Users/Ruud/Localhost/Site/extensions/local/Company/async/Extension.php" at line 20.' in /Users/Ruud/Localhost/Site/vendor/symfony/http-foundation/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php:144
Stack trace:
#0 /Users/Ruud/Localhost/Site/vendor/symfony/http-foundation/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php(262): Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->start()
#1 /Users/Ruud/Localhost/Site/vendor/symfony/http-foundation/Symfony/Component/HttpFoundation/Session/Session.php(81): Symfony\Component\HttpFoundation\Session\Storage\NativeSessionStorage->getBag('attributes')
#2 /Users/Ruud/Localhost/Site/vendor/bolt/bolt/src/Application.php(399): Symfony\Component\HttpFoundation\Session\Session->has('user')
#3 /Users/Ruud/Localhost/Site/vendor/bolt/bolt/src/Application.php(116): Bolt\A in /Users/Ruud/Localhost/Site/vendor/symfony/http-foundation/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php on line 144
```

**Comment by GawainLynch** [comment]
This is what I get from using your code:
![screenshot from 2015-02-01 10 06 18](https://cloud.githubusercontent.com/assets/1427081/5991735/01a59d72-a9fa-11e4-86a3-315a0a4b5e47.png)

I am thinking this is down to a php.ini or xdebug setting swallowing the errors somewhere.

**Comment by RuudBurger** [comment]
I think you're correct. It seems to work now using default php logging (was expecting bolt to catch the errors and display the Whoops page).
Thanks for your help!

**Comment by bobdenotter** [comment]
Fixed by #2655
